### PR TITLE
fix(hydration): narrow blast radius + fix Python import names (#178, #179)

### LINKED ISSUE #178: hydration.rs: caller blast radius triggers on any body edit, not signature changes
Surfaced by quorum self-review on PR for batch-3 hydration fixes.

`hydrate()` invokes `find_callers_of` whenever any line inside a function definition is in the changed range, not just when the function's signature line is changed. The intent is 'signature changed → find callers', but the implementation is 'any body edit → find callers'. Result: noisy caller-context injection on edits that don't change the function's external contract.

**Source:** src/hydration.rs (the for loop near line 70 that walks all_funcs and triggers find_callers_of on any overlap).

**Suggested fix:** narrow the trigger to changes that overlap the function's signature line (`fstart..=fstart` rather than `fstart..=fend`).

**Self-review by:** quorum v0.18.1 self-review on src/hydration.rs (HIGH, logic).

Use TDD: RED test asserting that body-only edits do NOT populate `ctx.callers`, then GREEN with the narrowed trigger.

### LINKED ISSUE #179: hydration.rs: Python import parsing returns wrong local names
Surfaced by quorum self-review on PR for batch-3 hydration fixes.

`extract_imported_names` Python branch (around src/hydration.rs:255) does not handle the local-binding asymmetry in Python imports correctly. Examples:

- `from x import foo as bar` should yield `['bar']` (the local binding) but probably yields `['foo']` (the imported name).
- `import foo.bar as baz` should yield `['baz']`.
- `from x import (a, b, c)` (parenthesized list) may not split correctly.

This is the Python analogue of #172 (Rust grouped use) and #173 (TypeScript default imports), both of which shipped in batch-3. Python was not in scope for batch-3; filing as follow-up.

**Source:** src/hydration.rs Python branch in extract_imported_names.

Use TDD: parameterized RED tests over the import forms above; GREEN with branch-aware parsing.

**Self-review by:** quorum v0.18.1 self-review on src/hydration.rs (HIGH, correctness).

### PR DESCRIPTION
**Pull request by jsnyder**
## Summary

- **#178**: Caller blast radius was triggering on any function body edit, not just signature changes. Narrowed the overlap condition from `*fstart <= end && *fend >= start` (any line in function) to `*fstart >= start && *fstart <= end` (signature line only). Filed #192 for the multiline-signature edge case as a follow-up enhancement.
- **#179**: Python import parsing returned wrong local names in two cases:
  - `from x import foo as bar` returned `foo` (before `as`) instead of `bar` (the local binding) — `.split(" as ").next()` should have been the last element
  - `import foo.bar as baz` returned `bar` (last dotted segment) instead of `baz` (the alias) — no `as` handling existed
  - Parenthesized `from x import (a, b, c)` now strips parens before splitting

## Test plan

- [x] `caller_blast_radius_ignores_body_only_edits` — body-only edit does NOT trigger callers
- [x] `caller_blast_radius_triggers_on_signature_edit` — signature edit DOES trigger callers
- [x] `caller_blast_radius_wide_body_edit_no_trigger` — multi-line body edit no trigger
- [x] `python_from_import_as_returns_local_binding` — `join as pjoin` yields `pjoin`
- [x] `python_from_import_parenthesized` — `(path, getcwd, listdir)` strips parens
- [x] `python_from_import_parenthesized_with_alias_and_trailing_comma` — `(path as p, getcwd,)`
- [x] `python_from_import_mixed_alias_and_plain` — `foo, bar as b` yields `[foo, b]`
- [x] `python_import_as_returns_alias` — `import foo.bar as baz` yields `baz`
- [x] `python_import_dotted_no_alias` — `import os.path` yields `path` (regression guard)
- [x] `python_import_simple_no_alias` — `import sys` yields `sys` (regression guard)

1756 tests passing, 0 failures.

## Quorum self-review

9 findings on `src/hydration.rs`, all pre-existing (complexity, ast-patterns). One new edge case (multiline signatures) filed as #192.

Closes #178, closes #179.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Caller blast-radius now scopes to functions whose signature lines fall within changed ranges, improving precision.

* **Refactor**
  * Improved Python import parsing to handle parenthesized import lists, trailing commas, comma-separated modules, and `as` aliases; returns the local binding name.

* **Tests**
  * Added tests for caller scoping behavior and expanded Python import syntax coverage.

### CODE REVIEW

**Comment by jsnyder** [comment]
Both findings addressed in e1d119e:

1. **`importlib` split bug**: Changed `text.split("import")" to `text.split(" import ")" — the space-delimited split won't match inside module names.
2. **Comma-separated bare imports**: Added comma-splitting loop in the `import X` branch, with per-part `as` alias handling and dotted-name fallback.

3 new tests added: `python_from_importlib_not_split_on_module_name`, `python_import_comma_separated_modules`, `python_import_comma_separated_dotted_with_alias`. Suite at 1759 passed, 0 failed.

File: src/hydration.rs [hunk 1]
```diff
@@ -309,11 +318,21 @@ fn extract_imported_names(import_text: &str) -> Vec<String> {
...
+                continue;
+            }
+            let name = if let Some(after_as) = part.split(" as ").nth(1) {
+                after_as.trim()
+            } else {
+                part.split('.').last().unwrap_or(part).trim()
+            };
+            if !name.is_empty() {
+                names.push(name.to_string());
+            }
```
[the remark this answers is not in the document]
Comment (jsnyder) [reply src/hydration.rs:335]: Won't fix. While Python's top-level binding for `import os.path` is indeed `os`, this function feeds into `import_targets` for Context7 doc enrichment. The last segment (`path`) is more useful for library doc lookup than the top-level namespace (`os`). This has been the behavior since the function was written, and `python_import_dotted_no_alias` guards it as a regression test.
